village/src/components/Smurfs.js

```diff
@@ -4,18 +4,26 @@ import Smurf from './Smurf';
 
 class Smurfs extends Component {
   render() {
+    let smurfs;
+    if(this.props.smurfs[0]&&this.props.smurfs[0]=='single'){
+      smurfs = this.props.location.state.smurf
+    }else{
```

Comment:
While this works, a better way to handle displaying only one component would be to either a. create a separate component for the smurf's individual page, or b. Have the list of smurfs that is passed down filter based on the url params.